packages/date-picker/src/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { forwardRef } from 'react';
+
+import LeafyGreenProvider, {
+  useDarkMode,
+} from '@leafygreen-ui/leafygreen-provider';
+import { BaseFontSize } from '@leafygreen-ui/tokens';
+import { useUpdatedBaseFontSize } from '@leafygreen-ui/typography';
+
+import {
+  contextPropNames,
+  DatePickerProvider,
+} from '../shared/components/DatePickerContext';
+import { useControlledValue } from '../shared/hooks';
+import { pickAndOmit } from '../shared/utils';
+
+import { DatePickerProps } from './DatePicker.types';
+import { DatePickerComponent } from './DatePickerComponent';
+import { SingleDateProvider } from './SingleDateContext';
+
+/**
+ * LeafyGreen Date Picker component
+ */
+export const DatePicker = forwardRef<HTMLDivElement, DatePickerProps>(
+  (
+    {
+      value: valueProp,
+      initialValue: initialProp,
+      onDateChange: onChangeProp,
+      handleValidation,
+      darkMode: darkModeProp,
+      baseFontSize: basefontSizeProp,
+      ...props
+    }: DatePickerProps,
+    fwdRef,
+  ) => {
+    const { darkMode } = useDarkMode(darkModeProp);
+    const baseFontSize = useUpdatedBaseFontSize(basefontSizeProp);
+    const [contextProps, restProps] = pickAndOmit(
+      { darkMode, baseFontSize, ...props },
+      contextPropNames,
+    );
+
+    const { value, setValue } = useControlledValue(
+      valueProp,
+      onChangeProp,
+      initialProp,
+    );
+
+    return (
+      <DatePickerProvider {...contextProps}>
+        <SingleDateProvider
+          value={value}
+          setValue={setValue}
+          handleValidation={handleValidation}
+        >
+          <LeafyGreenProvider
```

Comment:
It's possible to pass in a `darkMode` or `baseFontSize` prop into the component that doesn't match the outer context. In this case, we want to establish a new LGContext to avoid passing down the modified `darkMode`/`baseFontSize` value down multiple levels

Reply:
Got it! Missed that we'd need nested components to inherit the local override.